mountpoint-s3/src/inode.rs

```diff
@@ -1255,6 +1255,8 @@ pub struct InodeStat {
     pub etag: Option<String>,
     /// Storage class for the file (object), if known
     storage_class: Option<StorageClass>,
+    /// For objects with GLACIER or DEEP_ARCHIVE storage class shows if the object is restored (=> readable)
+    pub restored: bool,
```

Comment:
Now we're putting `restored` here, I think we can probably just delete `storage_class` and instead make this a single `is_readable` flag or something.

Reply:
This inclined me to [print](https://github.com/awslabs/mountpoint-s3/pull/467/files#diff-72de80783ffd916a6b6d2b704c855eadcb03d30e1d6315a0e33a30196cb7ca1aR1286) a warning on `InodeStat` instantiation. Previously it was printed from `new_read_handle ` or from `make_attr ` when listing a directory).

Generic `is_readable` implies that we may add new cases when the object has restricted permissions.